src/core/public/ui_settings/ui_settings_service.ts

```diff
@@ -40,9 +40,9 @@ export class UiSettingsService {
     this.uiSettingsApi = new UiSettingsApi(basePath, injectedMetadata.getKibanaVersion());
     loadingCount.add(this.uiSettingsApi.getLoadingCount$());
 
-    // TODO: when we have time to refactor the UiSettingsClient and all consumers
-    // we should stop using the legacy format and pick a better one
-    const legacyMetadata = injectedMetadata.getLegacyMetadata();
+    // TODO: Migrate away from legacyMetadata https://github.com/elastic/kibana/issues/22779
+    const legacyMetadata: any = injectedMetadata.getLegacyMetadata();
```

Comment:
Won't legacyMetadata at least be `{[key: string]: unknown}`? https://github.com/elastic/kibana/pull/22911/files#diff-e29dce8e241fa3347ae27b293da710c2R31

Reply:
Yeah, but `unknown` can only be assigned to `any` variables, so while I could do something like:

```ts
const legacyMetadata: {
  uiSettings: {
    defaults: UiSettingsState;
    user: UiSettingsState;
  };
} = injectedMetadata.getLegacyMetadata() as any;
```

Until we type the legacyMetadata or complete #22779 `any` is going to be involved somewhere, so it's kind of pointless IMO.

Reply:
Alright, I just went ahead and defined the types for `legacyMetadata`. The injected vars were really the only major thing in there that was scary to type (though I plopped in `unknowns` in for app, translations, and nav)